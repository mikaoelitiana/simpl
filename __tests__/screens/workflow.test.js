import React from "react";
import { mount } from "enzyme";
import Detail from "../../src/screens/workflow";

describe("Detail screen", () => {
  const rendered = mount(<Detail />)
  it("matches initial snapshot", () => {
    expect(rendered).toMatchSnapshot();
  });
})