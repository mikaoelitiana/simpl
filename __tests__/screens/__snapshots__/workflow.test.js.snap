// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail screen matches initial snapshot 1`] = `
<Workflow
  navigation={
    Object {
      "getParam": [MockFunction] {
        "calls": Array [
          Array [
            "item",
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": Object {
              "description": "A few minutes of workout in the afternoon to keep in good shape",
              "key": "2",
              "steps": Array [
                Object {
                  "description": "💪🏼",
                  "timer": 3,
                  "title": "Push-up",
                },
                Object {
                  "description": "⛹🏼‍♀️",
                  "timer": 20,
                  "title": "Jumping",
                },
              ],
              "title": "Afternoon workout",
            },
          },
        ],
      },
    }
  }
>
  <LinearGradient
    colors={
      Array [
        "#1A2980",
        "#26D0CE",
      ]
    }
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#F5FCFF",
        "flex": 1,
        "justifyContent": "center",
        "opacity": 0.9,
        "padding": 18,
        "paddingTop": 50,
      }
    }
  >
    <_class
      colors={
        Array [
          4279904640,
          4280733902,
        ]
      }
      locations={null}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#F5FCFF",
          "flex": 1,
          "justifyContent": "center",
          "opacity": 0.9,
          "padding": 18,
          "paddingTop": 50,
        }
      }
    >
      <BVLinearGradient
        colors={
          Array [
            4279904640,
            4280733902,
          ]
        }
        locations={null}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#F5FCFF",
            "flex": 1,
            "justifyContent": "center",
            "opacity": 0.9,
            "padding": 18,
            "paddingTop": 50,
          }
        }
      >
        <ScrollView>
          <_class>
            <RCTScrollView>
              <Component>
                <View>
                  <FlowItem
                    description="A few minutes of workout in the afternoon to keep in good shape"
                    key="2"
                    steps={
                      Array [
                        Object {
                          "description": "💪🏼",
                          "timer": 3,
                          "title": "Push-up",
                        },
                        Object {
                          "description": "⛹🏼‍♀️",
                          "timer": 20,
                          "title": "Jumping",
                        },
                      ]
                    }
                    title="Afternoon workout"
                  >
                    <TouchableOpacity
                      activeOpacity={0.2}
                      style={
                        Object {
                          "borderLeftColor": "#ffffff",
                          "borderLeftWidth": 3,
                          "marginBottom": 20,
                          "paddingLeft": 10,
                        }
                      }
                    >
                      <AnimatedComponent
                        accessible={true}
                        isTVSelectable={true}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Array [
                            Object {
                              "borderLeftColor": "#ffffff",
                              "borderLeftWidth": 3,
                              "marginBottom": 20,
                              "paddingLeft": 10,
                            },
                            Object {
                              "opacity": 1,
                            },
                          ]
                        }
                      >
                        <Component
                          accessible={true}
                          isTVSelectable={true}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Object {
                              "borderLeftColor": "#ffffff",
                              "borderLeftWidth": 3,
                              "marginBottom": 20,
                              "opacity": 1,
                              "paddingLeft": 10,
                            }
                          }
                        >
                          <View
                            accessible={true}
                            isTVSelectable={true}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Object {
                                "borderLeftColor": "#ffffff",
                                "borderLeftWidth": 3,
                                "marginBottom": 20,
                                "opacity": 1,
                                "paddingLeft": 10,
                              }
                            }
                          >
                            <CustomText
                              bold={true}
                              style={
                                Object {
                                  "color": "#ffffff",
                                  "fontSize": 18,
                                  "fontWeight": "bold",
                                }
                              }
                            >
                              <Text
                                accessible={true}
                                allowFontScaling={true}
                                ellipsizeMode="tail"
                                style={
                                  Array [
                                    Object {
                                      "fontFamily": "Ubuntu-Regular",
                                    },
                                    Object {
                                      "fontFamily": "Ubuntu-Bold",
                                    },
                                    Object {
                                      "color": "#ffffff",
                                      "fontSize": 18,
                                      "fontWeight": "bold",
                                    },
                                  ]
                                }
                              >
                                <Text
                                  accessible={true}
                                  allowFontScaling={true}
                                  ellipsizeMode="tail"
                                  style={
                                    Array [
                                      Object {
                                        "fontFamily": "Ubuntu-Regular",
                                      },
                                      Object {
                                        "fontFamily": "Ubuntu-Bold",
                                      },
                                      Object {
                                        "color": "#ffffff",
                                        "fontSize": 18,
                                        "fontWeight": "bold",
                                      },
                                    ]
                                  }
                                >
                                  Afternoon workout
                                </Text>
                              </Text>
                            </CustomText>
                            <CustomText
                              style={
                                Object {
                                  "color": "#ffffff",
                                  "fontSize": 14,
                                }
                              }
                            >
                              <Text
                                accessible={true}
                                allowFontScaling={true}
                                ellipsizeMode="tail"
                                style={
                                  Array [
                                    Object {
                                      "fontFamily": "Ubuntu-Regular",
                                    },
                                    null,
                                    Object {
                                      "color": "#ffffff",
                                      "fontSize": 14,
                                    },
                                  ]
                                }
                              >
                                <Text
                                  accessible={true}
                                  allowFontScaling={true}
                                  ellipsizeMode="tail"
                                  style={
                                    Array [
                                      Object {
                                        "fontFamily": "Ubuntu-Regular",
                                      },
                                      null,
                                      Object {
                                        "color": "#ffffff",
                                        "fontSize": 14,
                                      },
                                    ]
                                  }
                                >
                                  A few minutes of workout in the afternoon to keep in good shape
                                </Text>
                              </Text>
                            </CustomText>
                          </View>
                        </Component>
                      </AnimatedComponent>
                    </TouchableOpacity>
                  </FlowItem>
                  <Component
                    key="0"
                    style={
                      Object {
                        "flexDirection": "row",
                        "margin": 10,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flexDirection": "row",
                          "margin": 10,
                        }
                      }
                    >
                      <Component
                        style={
                          Object {
                            "marginHorizontal": 10,
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "marginHorizontal": 10,
                            }
                          }
                        >
                          <CustomText
                            style={
                              Array [
                                Object {
                                  "color": "#FFFFFF",
                                  "fontSize": 14,
                                },
                                Object {
                                  "borderColor": "#FFFFFF",
                                  "borderRadius": 15,
                                  "borderWidth": 1,
                                  "height": 30,
                                  "lineHeight": 26,
                                  "textAlign": "center",
                                  "textAlignVertical": "center",
                                  "width": 30,
                                },
                              ]
                            }
                          >
                            <Text
                              accessible={true}
                              allowFontScaling={true}
                              ellipsizeMode="tail"
                              style={
                                Array [
                                  Object {
                                    "fontFamily": "Ubuntu-Regular",
                                  },
                                  null,
                                  Array [
                                    Object {
                                      "color": "#FFFFFF",
                                      "fontSize": 14,
                                    },
                                    Object {
                                      "borderColor": "#FFFFFF",
                                      "borderRadius": 15,
                                      "borderWidth": 1,
                                      "height": 30,
                                      "lineHeight": 26,
                                      "textAlign": "center",
                                      "textAlignVertical": "center",
                                      "width": 30,
                                    },
                                  ],
                                ]
                              }
                            >
                              <Text
                                accessible={true}
                                allowFontScaling={true}
                                ellipsizeMode="tail"
                                style={
                                  Array [
                                    Object {
                                      "fontFamily": "Ubuntu-Regular",
                                    },
                                    null,
                                    Array [
                                      Object {
                                        "color": "#FFFFFF",
                                        "fontSize": 14,
                                      },
                                      Object {
                                        "borderColor": "#FFFFFF",
                                        "borderRadius": 15,
                                        "borderWidth": 1,
                                        "height": 30,
                                        "lineHeight": 26,
                                        "textAlign": "center",
                                        "textAlignVertical": "center",
                                        "width": 30,
                                      },
                                    ],
                                  ]
                                }
                              >
                                0
                              </Text>
                            </Text>
                          </CustomText>
                        </View>
                      </Component>
                      <Component>
                        <View>
                          <CustomText
                            style={
                              Array [
                                Object {
                                  "color": "#FFFFFF",
                                  "fontSize": 14,
                                },
                              ]
                            }
                          >
                            <Text
                              accessible={true}
                              allowFontScaling={true}
                              ellipsizeMode="tail"
                              style={
                                Array [
                                  Object {
                                    "fontFamily": "Ubuntu-Regular",
                                  },
                                  null,
                                  Array [
                                    Object {
                                      "color": "#FFFFFF",
                                      "fontSize": 14,
                                    },
                                  ],
                                ]
                              }
                            >
                              <Text
                                accessible={true}
                                allowFontScaling={true}
                                ellipsizeMode="tail"
                                style={
                                  Array [
                                    Object {
                                      "fontFamily": "Ubuntu-Regular",
                                    },
                                    null,
                                    Array [
                                      Object {
                                        "color": "#FFFFFF",
                                        "fontSize": 14,
                                      },
                                    ],
                                  ]
                                }
                              >
                                3
                                "
                              </Text>
                            </Text>
                          </CustomText>
                        </View>
                      </Component>
                      <Component
                        style={
                          Object {
                            "marginHorizontal": 10,
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "marginHorizontal": 10,
                            }
                          }
                        >
                          <CustomText
                            style={
                              Array [
                                Object {
                                  "color": "#FFFFFF",
                                  "fontSize": 14,
                                },
                                Object {
                                  "fontWeight": "bold",
                                },
                              ]
                            }
                          >
                            <Text
                              accessible={true}
                              allowFontScaling={true}
                              ellipsizeMode="tail"
                              style={
                                Array [
                                  Object {
                                    "fontFamily": "Ubuntu-Regular",
                                  },
                                  null,
                                  Array [
                                    Object {
                                      "color": "#FFFFFF",
                                      "fontSize": 14,
                                    },
                                    Object {
                                      "fontWeight": "bold",
                                    },
                                  ],
                                ]
                              }
                            >
                              <Text
                                accessible={true}
                                allowFontScaling={true}
                                ellipsizeMode="tail"
                                style={
                                  Array [
                                    Object {
                                      "fontFamily": "Ubuntu-Regular",
                                    },
                                    null,
                                    Array [
                                      Object {
                                        "color": "#FFFFFF",
                                        "fontSize": 14,
                                      },
                                      Object {
                                        "fontWeight": "bold",
                                      },
                                    ],
                                  ]
                                }
                              >
                                Push-up
                              </Text>
                            </Text>
                          </CustomText>
                          <CustomText
                            style={
                              Array [
                                Object {
                                  "color": "#FFFFFF",
                                  "fontSize": 14,
                                },
                              ]
                            }
                          >
                            <Text
                              accessible={true}
                              allowFontScaling={true}
                              ellipsizeMode="tail"
                              style={
                                Array [
                                  Object {
                                    "fontFamily": "Ubuntu-Regular",
                                  },
                                  null,
                                  Array [
                                    Object {
                                      "color": "#FFFFFF",
                                      "fontSize": 14,
                                    },
                                  ],
                                ]
                              }
                            >
                              <Text
                                accessible={true}
                                allowFontScaling={true}
                                ellipsizeMode="tail"
                                style={
                                  Array [
                                    Object {
                                      "fontFamily": "Ubuntu-Regular",
                                    },
                                    null,
                                    Array [
                                      Object {
                                        "color": "#FFFFFF",
                                        "fontSize": 14,
                                      },
                                    ],
                                  ]
                                }
                              >
                                💪🏼
                              </Text>
                            </Text>
                          </CustomText>
                        </View>
                      </Component>
                    </View>
                  </Component>
                  <Component
                    key="1"
                    style={
                      Object {
                        "flexDirection": "row",
                        "margin": 10,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flexDirection": "row",
                          "margin": 10,
                        }
                      }
                    >
                      <Component
                        style={
                          Object {
                            "marginHorizontal": 10,
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "marginHorizontal": 10,
                            }
                          }
                        >
                          <CustomText
                            style={
                              Array [
                                Object {
                                  "color": "#FFFFFF",
                                  "fontSize": 14,
                                },
                                Object {
                                  "borderColor": "#FFFFFF",
                                  "borderRadius": 15,
                                  "borderWidth": 1,
                                  "height": 30,
                                  "lineHeight": 26,
                                  "textAlign": "center",
                                  "textAlignVertical": "center",
                                  "width": 30,
                                },
                              ]
                            }
                          >
                            <Text
                              accessible={true}
                              allowFontScaling={true}
                              ellipsizeMode="tail"
                              style={
                                Array [
                                  Object {
                                    "fontFamily": "Ubuntu-Regular",
                                  },
                                  null,
                                  Array [
                                    Object {
                                      "color": "#FFFFFF",
                                      "fontSize": 14,
                                    },
                                    Object {
                                      "borderColor": "#FFFFFF",
                                      "borderRadius": 15,
                                      "borderWidth": 1,
                                      "height": 30,
                                      "lineHeight": 26,
                                      "textAlign": "center",
                                      "textAlignVertical": "center",
                                      "width": 30,
                                    },
                                  ],
                                ]
                              }
                            >
                              <Text
                                accessible={true}
                                allowFontScaling={true}
                                ellipsizeMode="tail"
                                style={
                                  Array [
                                    Object {
                                      "fontFamily": "Ubuntu-Regular",
                                    },
                                    null,
                                    Array [
                                      Object {
                                        "color": "#FFFFFF",
                                        "fontSize": 14,
                                      },
                                      Object {
                                        "borderColor": "#FFFFFF",
                                        "borderRadius": 15,
                                        "borderWidth": 1,
                                        "height": 30,
                                        "lineHeight": 26,
                                        "textAlign": "center",
                                        "textAlignVertical": "center",
                                        "width": 30,
                                      },
                                    ],
                                  ]
                                }
                              >
                                1
                              </Text>
                            </Text>
                          </CustomText>
                        </View>
                      </Component>
                      <Component>
                        <View>
                          <CustomText
                            style={
                              Array [
                                Object {
                                  "color": "#FFFFFF",
                                  "fontSize": 14,
                                },
                              ]
                            }
                          >
                            <Text
                              accessible={true}
                              allowFontScaling={true}
                              ellipsizeMode="tail"
                              style={
                                Array [
                                  Object {
                                    "fontFamily": "Ubuntu-Regular",
                                  },
                                  null,
                                  Array [
                                    Object {
                                      "color": "#FFFFFF",
                                      "fontSize": 14,
                                    },
                                  ],
                                ]
                              }
                            >
                              <Text
                                accessible={true}
                                allowFontScaling={true}
                                ellipsizeMode="tail"
                                style={
                                  Array [
                                    Object {
                                      "fontFamily": "Ubuntu-Regular",
                                    },
                                    null,
                                    Array [
                                      Object {
                                        "color": "#FFFFFF",
                                        "fontSize": 14,
                                      },
                                    ],
                                  ]
                                }
                              >
                                20
                                "
                              </Text>
                            </Text>
                          </CustomText>
                        </View>
                      </Component>
                      <Component
                        style={
                          Object {
                            "marginHorizontal": 10,
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "marginHorizontal": 10,
                            }
                          }
                        >
                          <CustomText
                            style={
                              Array [
                                Object {
                                  "color": "#FFFFFF",
                                  "fontSize": 14,
                                },
                                Object {
                                  "fontWeight": "bold",
                                },
                              ]
                            }
                          >
                            <Text
                              accessible={true}
                              allowFontScaling={true}
                              ellipsizeMode="tail"
                              style={
                                Array [
                                  Object {
                                    "fontFamily": "Ubuntu-Regular",
                                  },
                                  null,
                                  Array [
                                    Object {
                                      "color": "#FFFFFF",
                                      "fontSize": 14,
                                    },
                                    Object {
                                      "fontWeight": "bold",
                                    },
                                  ],
                                ]
                              }
                            >
                              <Text
                                accessible={true}
                                allowFontScaling={true}
                                ellipsizeMode="tail"
                                style={
                                  Array [
                                    Object {
                                      "fontFamily": "Ubuntu-Regular",
                                    },
                                    null,
                                    Array [
                                      Object {
                                        "color": "#FFFFFF",
                                        "fontSize": 14,
                                      },
                                      Object {
                                        "fontWeight": "bold",
                                      },
                                    ],
                                  ]
                                }
                              >
                                Jumping
                              </Text>
                            </Text>
                          </CustomText>
                          <CustomText
                            style={
                              Array [
                                Object {
                                  "color": "#FFFFFF",
                                  "fontSize": 14,
                                },
                              ]
                            }
                          >
                            <Text
                              accessible={true}
                              allowFontScaling={true}
                              ellipsizeMode="tail"
                              style={
                                Array [
                                  Object {
                                    "fontFamily": "Ubuntu-Regular",
                                  },
                                  null,
                                  Array [
                                    Object {
                                      "color": "#FFFFFF",
                                      "fontSize": 14,
                                    },
                                  ],
                                ]
                              }
                            >
                              <Text
                                accessible={true}
                                allowFontScaling={true}
                                ellipsizeMode="tail"
                                style={
                                  Array [
                                    Object {
                                      "fontFamily": "Ubuntu-Regular",
                                    },
                                    null,
                                    Array [
                                      Object {
                                        "color": "#FFFFFF",
                                        "fontSize": 14,
                                      },
                                    ],
                                  ]
                                }
                              >
                                ⛹🏼‍♀️
                              </Text>
                            </Text>
                          </CustomText>
                        </View>
                      </Component>
                    </View>
                  </Component>
                </View>
              </Component>
            </RCTScrollView>
          </_class>
        </ScrollView>
      </BVLinearGradient>
    </_class>
  </LinearGradient>
</Workflow>
`;
